src/main/java/fr/pilato/elasticsearch/crawler/fs/FsCrawlerImpl.java

```diff
@@ -568,30 +569,37 @@ private void indexFile(FileAbstractModel fileAbstractModel, ScanStatistic stats,
                     doc.getAttributes().setOwner(fileAbstractModel.owner);
                     doc.getAttributes().setGroup(fileAbstractModel.group);
                 }
-                // Attributes
-
+                // We index
                 if (fsSettings.getFs().isIndexContent()) {
-                    if (fsSettings.getFs().isJsonSupport()) {
+                    String fileExtension = FilenameUtils.getExtension(filename);
+                    if (fsSettings.getFs().isJsonSupport() && (fileExtension.equals("json") || fileExtension.equals("js"))) {
                         // https://github.com/dadoonet/fscrawler/issues/5 : Support JSon files
-                        esIndex(esClientManager.bulkProcessor(), fsSettings.getElasticsearch().getIndex(),
-                                fsSettings.getElasticsearch().getType(),
-                                generateIdFromFilename(filename, dirname),
-                                read(inputStream));
-                        return;
-                    } else if (fsSettings.getFs().isXmlSupport()) {
+                        if (fsSettings.getFs().isAddAsInnerObject()) {
+                            // https://github.com/dadoonet/fscrawler/issues/5 : Support JSon files
+                            doc.setObject(DocParser.fromJsonToMap(read(inputStream)));
+                        } else {
+                            esIndex(esClientManager.bulkProcessor(), fsSettings.getElasticsearch().getIndex(),
+                                    fsSettings.getElasticsearch().getType(),
+                                    generateIdFromFilename(filename, dirname),
+                                    read(inputStream));
+                            return;
+                        }
+                    } else if (fsSettings.getFs().isXmlSupport() && fileExtension.equals("xml")) {
```

Comment:
Same here. Don't test extension. If people want to mix different type of files, they can use `includes` parameter and launch multiple instances of FSCrawler.